projects/swimlane/ngx-graph/src/lib/graph/graph.component.ts

```diff
@@ -379,6 +382,11 @@ export class GraphComponent extends BaseChartComponent implements OnInit, OnChan
         this.tick();
       })
     );
+
+    if (this.graph.nodes.length === 0) {
+      return;
+    }
+
     result$.pipe(first(graph => graph.nodes.length > 0)).subscribe(() => this.applyNodeDimensions());
```

Comment:
I belive that I can remove the graph.nodes.length > 0 check here and just write ...first(graph)...